modules/core/src/main/java/org/apache/synapse/core/axis2/Axis2Sender.java

```diff
@@ -374,6 +375,13 @@ private static void doSOAPFormatConversion(org.apache.synapse.MessageContext syn
             // Hence, no conversion required is required
             if (!isResponseSOAP12 && !responseCtx.isDoingREST()) {
                 return;
+            } else {
+                try {
+                    // Message need to be serialized prior to the conversion
+                    MediatorPropertyUtils.serializeOMElement(synCtx);
```

Comment:
Shouldn't this happen after RelayUtils.buildMessage?

Reply:
Fixed.